app/src/main/java/mozilla/lockbox/support/Optional.kt

```diff
@@ -0,0 +1,10 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.lockbox.support
+
+data class Optional<T>(val value: T?)
+fun <T> T?.asOptional() = Optional(this)
```

Comment:
Ooh. Weird.

Reply:
gotta make Rx play nicely with optionals...

Reply:
Yeah, that sucks. Perhaps we shouldn't be using optionals in Rx?

Playing around, there don't seem to be any good answers to this.

Reply:
The concept of optionals in our observables is useful; we use the presence / absence of things like OAuthInfo to do routing in iOS and I'm trying to think of how it would look otherwise:

We could be more _explicit_ by having something like `val accountAvailable: Observable<Bool>` but it's another piece of state to update along with `oauthInfo` / `profile`. 

To avoid updating that state every time, we could just implement a filter for `oauthInfo == null`, which brings us back to the original problem of not being able to represent Optionals in Rx. I think we'll want them in other places so I'm OK with this solution for now.

At its core, this is a workaround for RxKotlin being implemented in Java and Java not having a concept of Optionals.